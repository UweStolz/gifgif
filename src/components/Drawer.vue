<template>
  <v-navigation-drawer
    color="red darken-3"
    mini-variant-width="8vh"
    mini-variant
    clipped
    nav
    app
  >
    <v-list>
      <v-list-item-group>
        <v-list-item
          v-for="(item, i) in items"
          :key="i"
          :to="item.link"
        >
          <v-list-item-icon>
            <v-badge
              :value="gifCount > 0"
              :content="gifCount"
              color="grey"
            >
              <v-icon
                color="white"
                large
                v-text="item.icon"
              />
            </v-badge>
          </v-list-item-icon>
        </v-list-item>
      </v-list-item-group>
    </v-list>
  </v-navigation-drawer>
</template>

<script lang="ts">
import { Vue, Component } from 'vue-property-decorator';
import { mdiHeart } from '@mdi/js';

@Component
export default class Drawer extends Vue {
    items = [
      {
        icon: mdiHeart, link: '/favourites',
      },
    ];

    gifCount: number = 0;

    async mounted() {
      this.gifCount = await this.$store.dispatch('getGifCount');
    }
}
</script>
