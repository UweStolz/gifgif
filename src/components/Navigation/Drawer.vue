<template>
  <v-navigation-drawer
    v-model="$store.state.showMobileDrawer"
    color="red darken-3"
    mobile-break-point="600px"
    mini-variant-width="8vh"
    mini-variant
    :permanent="!$vuetify.breakpoint.xsOnly"
    clipped
    app
  >
    <v-list>
      <v-list-item-group>
        <v-list-item
          v-for="(item, i) in items"
          :key="i"
          link
          :to="item.link"
        >
          <v-list-item-icon>
            <v-badge
              v-if="item.link === '/favourites'"
              :value="gifCount > 0"
              :content="gifCount"
              color="grey"
            >
              <v-icon
                color="white"
                large
                v-text="item.icon"
              />
            </v-badge>
            <v-icon
              v-else
              color="white"
              large
              v-text="item.icon"
            />
          </v-list-item-icon>
          <!-- Empty component for jumping list item on click -->
          <v-list-item-content />
        </v-list-item>
      </v-list-item-group>
    </v-list>
  </v-navigation-drawer>
</template>

<script lang="ts">
import { Vue, Component } from 'vue-property-decorator';
import { mdiHeart, mdiMagnify, mdiTranslate } from '@mdi/js';

@Component
export default class Drawer extends Vue {
  items = [
    {
      icon: mdiHeart, link: '/favourites',
    },
    {
      icon: mdiMagnify, link: '/search',
    },
    {
      icon: mdiTranslate, link: '/translate',
    },
  ];

  gifCount: number = 0;

  async mounted() {
    this.gifCount = this.$store.state.gifCount;
    this.$store.watch(
      () => this.$store.state.gifCount,
      (value: number) => {
        this.gifCount = value;
      },
    );
  }
}
</script>

<style>
span.v-badge__badge.grey {
  left: calc(100% - 15px) !important;
}
</style>
