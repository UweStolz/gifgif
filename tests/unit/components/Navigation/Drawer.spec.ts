import Drawer from '@/components/Navigation/Drawer.vue';
import shallow from '../../../helper';

describe('Drawer.vue', () => {
  it('renders properly', () => {
    const wrapper = shallow(Drawer);
    expect(wrapper).toMatchSnapshot();
  });
});
