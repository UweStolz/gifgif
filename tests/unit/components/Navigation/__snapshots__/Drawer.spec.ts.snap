// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer.vue renders properly 1`] = `
<v-navigation-drawer-stub app="true" color="red darken-3" clipped="true" height="100vh" minivariant="true" minivariantwidth="9vh" mobilebreakpoint="600px" permanent="true" src="" tag="nav" width="256">
  <v-list-stub tag="div" tile="true">
    <v-list-item-group-stub activeclass="v-item--active">
      <v-list-item-stub activeclass="" link="true" to="/favourites" tag="div">
        <v-list-item-icon-stub>
          <v-badge-stub color="grey" transition="scale-rotate-transition" content="0" label="$vuetify.badge">
            <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" right="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="16" nudgetop="0" nudgewidth="0" tag="span"> <span>Favourites</span></v-tooltip-stub>
          </v-badge-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub></v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" to="/search" tag="div">
        <v-list-item-icon-stub>
          <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" right="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="16" nudgetop="0" nudgewidth="0" tag="span"> <span>Tag search</span></v-tooltip-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub></v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" to="/translate" tag="div">
        <v-list-item-icon-stub>
          <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" right="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="16" nudgetop="0" nudgewidth="0" tag="span"> <span>Translate</span></v-tooltip-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub></v-list-item-content-stub>
      </v-list-item-stub>
    </v-list-item-group-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;
